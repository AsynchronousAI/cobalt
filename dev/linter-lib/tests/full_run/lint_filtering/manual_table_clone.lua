-- `manual_table_clone` changes its range if anything is in between the definition and loop,
-- but the lint filter should always be the same.

local stuff = {}

local output = {}

-- selene: allow(manual_table_clone)
for key, value in pairs(stuff) do
	output[key] = value
end

local other = {}
for key, value in pairs(stuff) do
	other[key] = value
end
