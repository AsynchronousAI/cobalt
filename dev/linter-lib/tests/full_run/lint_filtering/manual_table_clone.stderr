warning[manual_table_clone]: manual implementation of table.clone
   ┌─ manual_table_clone.lua:13:1
   │  
13 │ ╭ local other = {}
14 │ │ for key, value in pairs(stuff) do
15 │ │     other[key] = value
16 │ │ end
   │ ╰───^
   │  
   = try `local other = table.clone(stuff)`

